src/tcgwars/logic/impl/gen4/HeartgoldSoulsilverPromos.groovy

```diff
@@ -177,383 +198,258 @@ public enum HeartgoldSoulsilverPromos implements LogicCardInfo {
             energyCost G, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 20
+              flip { preventAllDamageNextTurn() }
             }
           }
-
         };
       case PLUSLE_HGSS16:
-        return basic (this, hp:HP060, type:LIGHTNING, retreatCost:1) {
-          weakness F
-          resistance M, MINUS20
-          move "Collect", {
-            text "Draw 2 cards."
-            energyCost C
-            attackRequirement {}
-            onAttack {
-              damage 0
-            }
-          }
-          move "Thunder Jolt", {
-            text "20 damage. Flip a coin. If tails, Plusle does 10 damage to itself."
-            energyCost L
-            attackRequirement {}
-            onAttack {
-              damage 0
-            }
-          }
-
-        };
+        return copy (Unleashed.PLUSLE_36, this);
       case MINUN_HGSS17:
-        return basic (this, hp:HP060, type:LIGHTNING, retreatCost:1) {
-          weakness F
-          resistance M, MINUS20
-          move "Call for Family", {
-            text "Search your deck for up to 2 Basic Pok�mon and put them onto your Bench. Shuffle your deck afterward."
-            energyCost C
-            callForFamily(basic:true, 2, delegate)
-          }
-          move "Tag Team Boost", {
-            text "10+ damage. If Plusle is on your Bench, this attack does 10 damage plus 20 more damage."
-            energyCost L
-            attackRequirement {}
-            onAttack {
-              damage 0
-            }
-          }
-
-        };
+        return copy (Unleashed.MINUN_34, this);
       case TROPICAL_TIDAL_WAVE_HGSS18:
-        return basicTrainer (this) {
-          text "Flip a coin. If heads, discard all Trainer and Stadium cards your opponent has in play. If tails, discard all Trainer and Stadium cards you have in play."
-          onPlay {
-          }
-          playRequirement{
-          }
-        };
+        return copy (NintendoBlackStarPromos.TROPICAL_TIDAL_WAVE_27, this);
       case RAIKOU_HGSS19:
         return basic (this, hp:HP080, type:LIGHTNING, retreatCost:1) {
           weakness F
           resistance M, MINUS20
           move "Zap Cannon", {
-            text "70 damage. During your next turn, Raikou can�t use Zap Cannon."
+            text "70 damage. During your next turn, Raikou can't use Zap Cannon."
             energyCost L, L, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 70
+              cantUseAttack(thisMove, self)
             }
           }
-
         };
       case ENTEI_HGSS20:
         return basic (this, hp:HP080, type:FIRE, retreatCost:1) {
           weakness W
           move "Flare Blitz", {
-            text "80 damage. Energy attached to Entei."
+            text "80 damage. Discard all [R] Energy attached to Entei."
             energyCost R, R, C, R
```

Comment:
The energy cost is wrong here. This was a quirk of many of the old gen 4 file templates and seemed to fail whenever there was an energy symbol in a text box.